nautobot/docs/installation/nautobot.md

```diff
@@ -0,0 +1,255 @@
+
+Many common issues can be solved by running the latest version of Pip. Before continuing with installing Nautobot, upgrade Pip to its latest release:
+
+```no-highlight
+$ sudo pip3 install --upgrade pip
+```
+
+## Choose your `NAUTOBOT_ROOT`
+
+This is where everything related to Nautobot will be installed. We're going to use this value across the documentation. You'll need to set the `NAUTOBOT_ROOT` environment variable to tell Nautobot where to find its files and settings at `/opt/nautobot`.
+
+```no-highlight
+$ export NAUTOBOT_ROOT=/opt/nautobot
+```
+
+## Create the Virtual Environment
+
+A Python [virtual environment](https://docs.python.org/3/tutorial/venv.html) or *virtualenv* is like a container for a set of Python packages. A virtualenv allows you to build environments suited to specific projects without interfering with system packages or other projects. When installed per the documentation, Nautobot uses a virtual environment in production.
+
+We're going to create the virtualenv as our `NAUTOBOT_ROOT` as the root user to bootstraps the `/opt/nautobot` directory and populate it with a self-contained Python environment. 
+
+```no-highlight
+$ sudo python3 -m venv $NAUTOBOT_ROOT
+```
+
+Next, change ownership of `NAUTOBOT_ROOT` to the `nautobot` user:
+
+```no-highlight
+$ sudo chown -R nautobot:nautobot $NAUTOBOT_ROOT
+```
+
+## Sudo to nautobot
+
+Now that we've created the virtualenv, the remaining steps will be performed as the `nautobot` user. 
+
+!!! warning
+    Don't skip this step!!
+
+    It's important to install Nautobot as the `nautobot` user so that we don't have to worry about fixing permissions later.
+
+```no-highlight
+$ sudo su - nautobot
+```
+
+After becoming `nautobot`, we need to set the `NAUTOBOT_ROOT` environment variable again for this user (since it is a fresh session as far as the system is concerned):
+
+```no-highlight
+$ export NAUTOBOT_ROOT=/opt/nautobot
+```
+
+### Add `NAUTOBOT_ROOT` to `.bashrc`
+
+For bonus points, add this to `~/.bashrc` for `nautobot`:
+
+```no-highlight
+$ echo "export NAUTOBOT_ROOT=/opt/nautobot" >> ~/.bashrc
+```
+
+From here on out, anytime you become `nautobot`, your `NAUTOBOT_ROOT` will be set automatically.
+
+## Activate the Virtual Environment
+
+!!! warning
+    This and all remaining steps in this document should all be performed as the `nautobot` user!
+
+    Hint: Use `sudo su - nautobot`
+
+To work inside a Python virtualenv, it must be activated. This makes sure that the version of Python you're using, as well any dependencies that you install remain isolated in this environment. 
+
+If it helps, try to think of activating the virtualenv like entering a shell unique to Nautobot's application environment.
+
+```no-highlight
+$ source /opt/nautobot/bin/activate
+(nautobot) $
+```
```

Comment:
We can do this automatically with:

```
echo ". /opt/nautobot/bin/activate" >> ~/.bashrc
```

This might help users from making stupid mistakes.  But also just because we add things to ~/.bashrc it doesn't mean for example systemd units will get those values as they won't source the bashrc file

Reply:
Yeah I don't want to do that yet. If we add `/opt/nautobot/bin` to `PATH`, there's never a need to activate the virtualenv. I want to tackle that in the next iteration!

Reply:
cool